### DISCUSSION TITLE
Any clues on finding those IDs?

### DISCUSSION THREAD

**Discussion by Jasonrobot00** [nirnachmani/Pixie-Plus-for-Home-Assistant#5] [Q&A]
Hi OP and community,

Is anyone able to provide examples on how this info should appear in Proxyman? Screenshots or text dump... I'll take anything! I'm trying Proxyman on Windows and iOS and not coming up with much.

Cheers

**Comment by nirnachmani** [comment]
Sorry, I have never used Proxyman. However, you should see https requests and answers, including from the pixie app. I did ask SAL to allow users to access those values through their app but it didn't go anywhere. Maybe if other users ask them they will make this info available in the app.

**Reply by Jasonrobot00** [reply]
Could you maybe provide me with an example of what the respective ids and key may look like just to see if I’m on the right path

**Reply by robothomessydney** [reply]
hey man! thanks for your hard work. FInding the Application ID and Installer ID is really simple using Proxyman. But we are stuck with finding the javscript key. Could you please kindly help :) we are stuck!

**Comment by brendanmckenzie** [comment]
I had trouble with Proxyman on iOS as well.  The PIXIE PLUS app refused to route traffic via the proxy.

I ended up using [mitm](https://mitmproxy.org/) on my Mac in WireGuard mode.  Installed WireGuard and the mitm SSL certificate on my iPhone.  With that in place, everything on my iPhone was routed through mitm and I was able to find the values.

It looks like the auth flow might have changed a little bit, though.  The javascriptkey seems to have been replaced with and "client key", and those values are passed as headers instead of parameters in the HTTP request JSON bodies.  I'm working on updating the code and will push a PR when it's ready.

Side note: from a quick test it looks as though installation ID can be set to anything, application ID and client key must be obtained from the app.

**Reply by nirnachmani** [reply]
Thanks. I also used a mitm with a virtual android device. Everything is still working well for me with the current code - I hope changing the code won't break anything.

**Comment by robothomessydney** [comment]
Hey Brendan,

Thanks a lot. Eager to get pixie across locally to HA!  how has this
integration been going for you?

**Reply by brendanmckenzie** [reply]
Ha, I literally just got it working after a whole morning of tinkering.  There seemed to be some API updates so I had to adapt the code a bit to accommodate.

But so far so good, I've got all my lights in Apple Home now which was the goal!

I've opened a PR with the changes.

**Comment by hoskerism** [comment]
Hi All,

I have been working through the steps to get the IDs - but have now hit a brick wall. Any help would be appreciated!

What is working:
Android AVD with root access via Magisk.
Mitm proxy installed via Magisk module in System Certificate store.
And I can capture https traffic from (almost) all apps via mitmweb.

However, the Pixie Plus app is refusing to connect. I get:
_Client TLS handshake failed. The client disconnected during the handshake. If this happens consistently for www.pixie.app, this may indicate that the client does not trust the proxy's certificate._

Any clues how to solve this? Has Pixie explicitly updated their app to stop people from obtaining the required IDs?

Any help appreciated!

**Reply by brendanmckenzie** [reply]
My issue was on iPhone where the PIXIE app wouldn't route traffic through the MITM proxy.

It worked when I used the Wireguard mode of MITM.

Your problem sounds like it might be the certificate, though.  If you use other things (like the web browser) do they work OK?

**Reply by nirnachmani** [reply]
@hoskerism, I do wonder if your issue is with the certificate. The traffic you capture is possibly only traffic that does not require a certificate. Maybe try to re-install the certificate? It's been long time since I have done that process and I no longer have the setup (AVD, mitm, etc) installed. Unfortunately, I don't currently have the time to replicate it. It is possible that Pixie updated the app to ignore system certificate - maybe download an older version of the app and try? Saying that, it seems that @brendanmckenzie has recently managed to find those parameters, so it is less likely to be the issue. Good luck!

**Reply by hoskerism** [reply]
Thanks both for your help. I decided to give up on the AVD and try @brendanmckenzie's iOS method using an old ipad (which I had previously given up on as it was too old to install Proxyman). The WireGuard method seemed to work like a charm!

In case it helps anyone else, the steps I used were as follows. (Apologies if I missed anything important):
1. Install Pixie Plus and WireGuard on iPad.
2. Install mitmproxy on Windows.
3. Run mitmweb in 'Normal' mode. Eg: from command prompt, type: mitmweb
4. On iPad, set up the proxy settings on the wifi network to use the PC running mitmproxy's IP address, and the default port of 8080.
5. On iPad, browse to mitm.it using Safari. Download the iOS certificate, install it and trust it using the instructions provided on the download page.
6. Restart mitmproxy, this time in WireGuard mode. Eg: from the command prompt, type: mitmweb --mode wireguard
7. On iPad, start the WireGuard app and create a new tunnel using the QR code displayed on the mitmproxy webpage. Enable the tunnel. (You should now start to see traffic being captured on the mitmproxy webpage).
8. On iPad, log in to the Pixie Plus app.
9. Inspect the traffic for www.pixie.app on the mitmproxy webpage. It is fairly easy to find the applicationId, clientKey and installationId.

Note that I haven't yet tried using these parameters on the Home Assistant integration, as I haven't yet installed it on my HA server. But that is a problem for another day.

Thanks again!

**Reply by hierrosh** [reply]
I followed @hoskerism method and end up with got the "applicationId" , "sessionToken" and "clientKey" in a WebSocket data, but I tried fill in ApplicationID with "applicationId", InstallationID with "sessionToken"  and JavascriptKey with "clientKey" but not work. Should I look for anything else?

**Reply by robotsmarthomes** [reply]
This works . Try again. Literally did an installation today using it

**Reply by hierrosh** [reply]
what did you get via mitm, "applicationId" , "sessionToken" and "clientKey" ? I failed so many times and I confirm my email address and password is correct.

**Reply by kim0sabe** [reply]
> Thanks both for your help. I decided to give up on the AVD and try @brendanmckenzie's iOS method using an old ipad (which I had previously given up on as it was too old to install Proxyman). The WireGuard method seemed to work like a charm!
> 
> 
> 
> In case it helps anyone else, the steps I used were as follows. (Apologies if I missed anything important):
> 
> 1. Install Pixie Plus and WireGuard on iPad.
> 
> 2. Install mitmproxy on Windows.
> 
> 3. Run mitmweb in 'Normal' mode. Eg: from command prompt, type: mitmweb
> 
> 4. On iPad, set up the proxy settings on the wifi network to use the PC running mitmproxy's IP address, and the default port of 8080.
> 
> 5. On iPad, browse to mitm.it using Safari. Download the iOS certificate, install it and trust it using the instructions provided on the download page.
> 
> 6. Restart mitmproxy, this time in WireGuard mode. Eg: from the command prompt, type: mitmweb --mode wireguard
> 
> 7. On iPad, start the WireGuard app and create a new tunnel using the QR code displayed on the mitmproxy webpage. Enable the tunnel. (You should now start to see traffic being captured on the mitmproxy webpage).
> 
> 8. On iPad, log in to the Pixie Plus app.
> 
> 9. Inspect the traffic for www.pixie.app on the mitmproxy webpage. It is fairly easy to find the applicationId, clientKey and installationId.
> 
> 
> 
> Note that I haven't yet tried using these parameters on the Home Assistant integration, as I haven't yet installed it on my HA server. But that is a problem for another day.
> 
> 
> 
> Thanks again!

This helped me heaps
Thank you

**Comment by robothomessydney** [comment]
If your password isn’t private

I’m happy for you to send me details and I can attempt to get credentials
for you

***@***.***

<https://www.robotsmarthomes.com.au/>
Jacob Russo | Project Manager
                         Client Relations Manager
e: ***@***.***
w: www.robotsmarthomes.com.au <http://www.robotsmarthomes.com.au/>
m: 0425 305 936
*a:  *Shop 2/174 Liverpool Road Enfield 2136

**Reply by hierrosh** [reply]
All done thank you very much for everyone, I found the problem because of I didn't use the latest patch, the JavaScriptKey needs to change to clientKey. Problem solved and all pixie switchs in HomeAssistant now.

**Comment by michealc** [comment]
Thanks everyone for assisting here.  I tried getting traffic from iOS app to macOS with proxyman without success. I could get googleapis but no www.pixie.app traffic.   As you can integrate pixie into smartthings, I removed and added the pixie intergration which provided a login window.  This time I could capture two POST api calls /oauth/authoize and /oauth/selectHome.  Here are the sanitised responses.  I can match up the requirements of applicationID, InstallationID, and client-key.  I tried multiple times and no combinations worked - unauthoised in adding integration.  Any pointers appreciated.

```
POST #1
https://www.pixie.app/p0/oauth/authorize?**uid=**bWN1bW1pb...&**upwd=**Y2tx...=&postTarget=authorize&response_type=code&**client_id=**d00a8...&state=eyJhbG...&redirect_uri=https://c2c-ap.smartthings.com/oauth/callback
RESPONSE
**uid=**bWN1...==&upwd=22aa8...&postTarget=authorize&response_type=code&**client_id=**d00a81599ad...&state=eyJhbG...&redirect_uri=https://c2c-ap.smartthings.com/oauth/callback&isSuccess=1&selectHome=1&homeList=[{"id":"b93f4c179ff...","name":"House","isOwner":true,"isHaveDevs":true,"isOnline":false,"isHaveGateway":true},{"id":"c3f6ef8e71e...","name":"Shed","isOwner":true,"isHaveDevs":true,"isOnline":true,"isHaveGateway":true}]&useId=ced4809...&session=5df3efb7...

POST #2 
https://www.pixie.app/p0/oauth/selectHome?useId=ced480918...&upwd=22aa8f57d4b4e...&session=5df3efb756f4...&selectHomeIds=[%22c3f6ef8e...%22]&postTarget=selectHome&response_type=code&client_id=d00a81599...&state=eyJhbGci...&redirect_uri=https://c2c-ap.smartthings.com/oauth/callback
RESPONSE
state=eyJhbGciOiJIU...&code=be917512994bfc...

```

**Reply by tomrobbo** [reply]
Yeah I could never get Proxyman to work.  I followed the above [comment](https://github.com/nirnachmani/Pixie-Plus-for-Home-Assistant/discussions/5#discussioncomment-7122696) (using macOS instead of windows) I got then easily
